### PR TITLE
Upload and download `.enso-project`s from the local backend (PM backend)

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description

- Closes https://github.com/enso-org/cloud-v2/issues/478
  - Download local project as `.enso-project` archive
    - Requires latest nightly version of Project Manager
- Closes https://github.com/enso-org/cloud-v2/issues/510
  - Allow uploading `.enso-project` to local backend
- Closes https://github.com/enso-org/cloud-v2/issues/477
  - Promote local project to cloud
    - Currently errors with 500 (when uploading a small bundle) or 413 (when uploading a large bundle). May be fixed soon

### Important Notes
The "upload project to cloud" context menu action does not currently have an entry in the new context menu, so that will probably need an official design at some point

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by somebody1234** [comment]
(requesting review while marked as draft, because the only issue *should* be on the PM side)

**Comment by xvcgreg** [comment]
@somebody1234 please take look here.

**Comment by somebody1234** [comment]
@xvcgreg I'll mark as draft temporarily, since there are more significant changes upcoming in https://github.com/enso-org/enso/pull/7377

I think it'd be better to wait until that gets merged in, before fixing conflicts and reopening this PR

**Comment by somebody1234** [comment]
@PabloBuchu i think this PR is ready for testing btw

**Comment by PabloBuchu** [comment]
When dropping `.enso-project` archive and after successful response with created project could we add new project to the table rather than file like? after reload it changes to project. I updated the response["project"] to have the same attributes as create_project 

Screenshot 2023-08-03 at 12 45 15

**Comment by PabloBuchu** [comment]
File upload (any not only `.enso-project`) is sending `--data-raw` and base64 file content making it corrupted when saving in s3. 

Can we use raw binary body? Working curl example
```
curl --location 'https://7aqkn3tnbc.execute-api.eu-west-1.amazonaws.com/files?file_name=KMeans_32.enso-project&parent_directory_id=directory-2TT9VR2uD0p6xwNDGR0iLYWuQRS' \
--header 'Authorization: Bearer eyJraWQiOiJiVjd1ZExrWTkxU2lVUWRpWVhDSVByRitoSTRYVHlOYTQ2TXhJRDlmY3EwPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiJjMjk1NWUzZC05OTExLTQzMDYtYjZjMS00MTQ1YWJlZTc4MzEiLCJpc3MiOiJodHRwczpcL1wvY29nbml0by1pZHAuZXUtd2VzdC0xLmFtYXpvbmF3cy5jb21cL2V1LXdlc3QtMV85S3ljdTJTYkQiLCJ2ZXJzaW9uIjoyLCJjbGllbnRfaWQiOiI0ajliZnM4ZTc0MTVlcmY4MmwxMjl2MHFoZSIsIm9yaWdpbl9qdGkiOiIxMjg4Y2M4OS1hZmVlLTRkMGEtOWFkOS04NmYzNzAyNTYxZmQiLCJ0b2tlbl91c2UiOiJhY2Nlc3MiLCJzY29wZSI6ImF3cy5jb2duaXRvLnNpZ25pbi51c2VyLmFkbWluIG9wZW5pZCBlbWFpbCIsImF1dGhfdGltZSI6MTY5MTA1NjIyMCwiZXhwIjoxNjkxMDY2MTI0LCJpYXQiOjE2OTEwNjI1MjQsImp0aSI6IjVjOTU4YzQ1LTIzZTUtNDE4NC1hY2Y2LTVhOWI3MjM3OWMyYiIsInVzZXJuYW1lIjoicGF3ZWwuYnVjaG93c2tpQGVuc28ub3JnIn0.FWe-dzPxjY1g-sDdt-wTNfGfJ-kSYzlVtaxZb1uq2tmEPyQJGq_Wyq8IqalpcdCkP9aVEDf-CVgpOVylVvcaj_gAx0DZeUcOs2l6SCy-ulx4D_yMukR7x_J7t8HOYefRE-oOHdo1mCxetAE9fmCSzIyyZZ8B8JGV4ytgQbbCcDMtn3orMCrQNEsTp8rj0UHSsQPQ9IC3NegnC6ZXK76EZpYgiUd7_C04umwwrm9KAgk67KUPqZOfphnaxV34fXWvpbYOO_UW97WhixuBCFLL036tEbZVMihGB6JqFf2NYTrGWLVgJRTGq8yOzdfmZxdkzunNKkWAOP3PzbJdhQwVnQ' \
--header 'Content-Type: application/octet-stream' \
--data '@/Users/pawelbuchowski/Desktop/KMeans_1.enso-project'
```

**Comment by somebody1234** [comment]
oh yeah. i have no clue why i thought i needed to post using `base64` tbh... should be a simple fix though

**Comment by somebody1234** [comment]
@PabloBuchu btw `response["project"]` should probably have `camelCase` properties?

**Comment by somebody1234** [comment]
pushed potential fixes, but they are :warning: not tested yet :warning:!
i'll probably test them tomorrow

**Comment by somebody1234** [comment]
basic testing:
note: this is not 100% thorough, would definitely be nice to have a bit more testing

- :heavy_check_mark: promote project to cloud uploads to cloud
- :grey_question: promoting project that has been edited works after uploading to cloud
- :heavy_check_mark: downloading single project works on local PM
- :heavy_check_mark: downloading multiple projects works on local PM
- :heavy_check_mark: can download running project
- :heavy_check_mark: can reupload downloaded local project to local PM
  - :heavy_check_mark: name updates properly
- :heavy_check_mark: can reupload downloaded local project to cloud
  - :warning: when it was broken, the failing upload was repeatedly retried. i think #7430 should fix this, but it will be good to confirm this. it's out of scope for this test, but i may test this later. this can be tested by trying to upload a file without an extension.
  - :warning: to open a newly uploaded project, you currently need to refresh. this is because the frontend assumes the returned project details are in `camelCase` rather than `snake_case` as it is currently. this is intentional; because the backend should be switching to `camelCase` soon
- :grey_question: uploading manually created project to cloud works
- :heavy_check_mark: uploading file to cloud still works correctly
- :heavy_check_mark: uploading a mix of files + projects adds the assets, and in the right places
- :x: opening uploaded cloud project works - status is returned as "opened" but `POST /resources` always returns `400`, no matter how long you wait
  - after some debugging, it seems like the cloud doesn't understand the format sent by the Project Manager, which has a top level folder inside the compressed bundle, rather than having all the project contents at the top level.
  - manually creating an archive without this nesting, seems to let the project open, however the Language Server endpoints seem to not exist. this is probably because the archive lacks the `.enso` VCS directory - this can likely be solved by importing the project into the Project Manager on the VM, which should create the `.enso` directory

**Comment by somebody1234** [comment]
@PabloBuchu there seem to be some backend issues with uploaded projects (see the last bullet point) - you might want to take a look at those

**Comment by somebody1234** [comment]
(merged to pull in some fixes)

**Comment by PabloBuchu** [comment]
ok so when I download and upload a project to cloud it works (the proper files are being send to s3) but when I use `Upload to cloud` I get 404 with a success (the project is being created properly)
Screenshot 2023-08-06 at 16 45 51

**Comment by somebody1234** [comment]
@PabloBuchu oops, the url it's trying to use is wrong. i've pushed a commit that should fix this

**Review by PabloBuchu — APPROVED** [review]

**Review by farmaazon — APPROVED** [review]
Fix docs before merging.

File: app/gui/config.yaml
Comment (farmaazon) [inline app/gui/config.yaml]: I don't see any `projectManagerHttpEndpoint` constant in `constants.rs.

Comment (PabloBuchu) [reply app/gui/config.yaml]: thanks good catch

Comment (somebody1234) [reply app/gui/config.yaml]: good catch, thanks. yeah, i just copied the docs from the entry above

**Pull request merged**
